### ISSUE TITLE
[FIX] [10.0] fix page choose function in the menu editor window

### ISSUE THREAD

**Issue by tsabi** [odoo/odoo#14922]
@beledouxdenis please look at his PR, as the original commit a98d60cb11ceee31a6bd9d3734d0e4b565e5d677 was yours.

Impacted versions: 10.0 saas6

Steps to reproduce: open the menu editor window, you cannot search to pages there

Current behavior: if you search for a page in the menu edit window, two error happens: odoo runs to error (because the context mistypo) and the page_exists always returns False (even if the page really exists or not).

Expected behavior: After my PR both error fixed

**Comment by beledouxdenis** [comment]
Hi @tsabi 

Actually, the fix was mean for Odoo 9.0 and there it was correct.

The typo comes from the forward-port to 10.0. I just solved with from b228f548ca10e61d3b802fa624bdaf8a2f8b11b2

Thanks for the report.

However, it looks like the rest of your PR is not mandatory to have the feature working: I just corrected the typo, and it worked at my end. In which case is it needed ?

**Comment by tsabi** [comment]
Hi @beledouxdenis!

If you look into my commit a822d3322f6d0db2997208c06d54157cbdf14bc6 i explained well what i fixed.

I debugged what ajax requests do the javascript code behind the menu editor window, and it do two calls in this order:
1. called the page_exists method, than
2. called the search_pages.

Both of them had (still have) errors.
The second one was fixed by your commit a98d60cb11ceee31a6bd9d3734d0e4b565e5d677 with a sightly modification with your new commit b228f548ca10e61d3b802fa624bdaf8a2f8b11b2. In my PR i had this also, and so a little bit disappointed you didn't comitted my PR, so i am not credited..

But the first one is still have bugs in the live code, and this is what i fixed in my a822d3322f6d0db2997208c06d54157cbdf14bc6 commit.

## Lets look detailed:
### The ajax request parameters was shifted off
If you look the ajax call page_exists how defined during the API conversion was not converted well. It have this code:
```javascript
    page_exists: function (term) {
        return this.call('page_exists', [null, term], {
            context: base.get_context(),
        });
    },
    fetch_pages: function (term) {
        return this.call('search_pages', [null, term], {
            limit: 100,
            context: base.get_context(),
        });
    },
```

```python
    @api.model
    def page_exists(self, name, module='website'):
```

The parameter ```null``` goes to ```name``` and ```term``` goes to ```module```. This is because page_exists method have the ```@api.model``` decorator instead of ```@api.multi``` like the search_pages have:

```python
    @api.multi
    def search_pages(self, needle=None, limit=None):
```
 In the ```fetch_pages``` javascript code the ```null``` goes into the ```ids``` what is fetched by the ```@api.multi``` decorator, and ```term``` goes to the ```needle```. This is so good.

Here we have two fix options:
1. change the decorator to ```@api.multi``` instead of ```@api.model```, or
2. remove the null from the javascript call.

I choosed the latter one.

### page_exists module parameter needs to check after the call (it could run to error if we provide None to module)

I changed the default value of module parameter variable to be None, and check later if it is None, than the default is ```'website'```
I did this because if i add a parameter from javascript as null, than the python code will goes to exception.

### the check "self.env.ref('%s.%s' % module, name)" have syntax error, it should be: "self.env.ref('%s.%s' % (module, name))"

This code:
```python
self.env.ref('%s.%s' % module, name)
```

always generate an exception. The correct form is:
```python
self.env.ref('%s.%s' % (module, name))
```

. But because it is inside a try except you never see this.
This page_exists method just simply returns False all the time!

### BUT it still completely wrong, as the web pages are not in ir.model.data, but in ir.ui.view. So lets refactor the search

And after i fixed the syntax error inside the try except, i realized it was not just syntaxically wrong, but logically also, so i wrote the correct code.
```self.env.ref``` looks in the ir.model.data table, while the web pages are in the ir.ui.view table. It don't need so much to accept this as a logical bug.

## Conclusion

After you merge my commit you will have a fully working page_exists method (i debugged it with ajax calls from javascript, now it works as expected).

And to answer jour question, why to fix bugs? I really need to answer to this?

In the javascript code behind the menu editor window, there are two calls of the page_exists ajax call, i am sure they are there with good reason.
At the moment with the live code the page_exists always reply with False, so we have two choices:
1. remove the call of page_exists completely (really there isn't any reason to have a ajax call what always returns False, it generates network noise), and the codes in the true conditions, becase they are impossible to run at any time.
2. Fix the page_exists method.

Let me ask back, why we leave a lot of bugs in the Odoo code?
